front/script.js

```diff
@@ -56,105 +61,123 @@ const fadeIn = (el, display = "flex") => {
-const showPraiseForm = () => {
-  playSound("next");
+const transitionToFormView = () => {
   fadeOut(elems.main);
   setTimeout(() => {
     fadeIn(elems.form);
     elems.sendBtn.classList.add("show");
+    // 文字数制限を開始
     updateCharCount();
     elems.input.addEventListener("input", updateCharCount);
   }, 600);
 };
 
+const transitionToEndView = () => {
+  // 上乗せした背景を消して初期背景に戻す
+  elems.bg.style.opacity = 0;
 
-// 文字数制限関係
-const MAX_LENGTH = 20;
-
-const updateCharCount = () => {
-  const count = elems.input.value.trim().length;
-  const counterEl = document.getElementById("charCount");
-  counterEl.innerText = `${count}/${MAX_LENGTH}`;
-  if (count > MAX_LENGTH) {
-    counterEl.style.color = "red";
-  } else {
-    counterEl.style.color = "#666";
-  }
+  // ありがとう を ２秒後 に表示
+  const thankYou = document.createElement("div");
+  thankYou.className = "thank-you";
+  thankYou.innerText = "ありがとう！";
+  document.body.appendChild(thankYou);
+  setTimeout(() => {
+    thankYou.style.opacity = 1;
+  }, 2000);
 };
 
-// 褒め言葉を送信
-const sendPraise = () => {
-  const text = elems.input.value.trim();
-
-  if (!text) {
-    alert("褒め言葉を入力してください。");
-    return;
-  }
-  if (text.length > MAX_LENGTH) {
-    alert(`褒め言葉は${MAX_LENGTH}文字以内でお願いします。`);
-    return;
-  }
-  if (text.includes(",")) {
-    alert("カンマ（,）は使えません。全角の「，」をご使用ください。");
-    return;
-  }
-
-  playSound("post");
+const showFallingText = (text) => {
   elems.form.classList.replace("fade-in", "fade-out");
   elems.form.classList.add("hidden");
 
-  const inputRect = elems.input.getBoundingClientRect();
-
   const fallingText = document.createElement("div");
-  fallingText.className = "praise-text-drop";
+  fallingText.className = "home-text-drop";
   fallingText.innerText = text;
+  const inputRect = elems.input.getBoundingClientRect();
   fallingText.style.position = "fixed";
   fallingText.style.top = `${inputRect.top + inputRect.height / 2}px`;
   fallingText.style.left = `${inputRect.left + inputRect.width / 2}px`;
   fallingText.style.transform = "translate(-50%, -50%) scale(1)";
   document.body.appendChild(fallingText);
+};
 
-  const thankYou = document.createElement("div");
-  thankYou.className = "thank-you";
-  thankYou.innerText = "ありがとう！";
-  document.body.appendChild(thankYou);
+// メイン処理 //
 
-  elems.bg.style.opacity = 0;
-  setTimeout(() => {
-    thankYou.style.opacity = 1;
-  }, 2000);
+// 褒め言葉を表示
+const showHome = async () => {
+  // 褒め言葉を取得してセット
+  elems.message.innerText = await getHome();
+  // 効果音
+  playSound("get");
+  // 初期画面 -> 褒め言葉表示画面 に遷移
+  transitionToHomeView();
+};
 
-  apiFetch("/homes", {
+// 褒めフォームを表示
+const showHomeForm = () => {
+  // 効果音
+  playSound("next");
+  // 褒め言葉表示画面 -> 褒め言葉入力画面 に遷移
+  transitionToFormView();
+};
+
+// 褒め言葉を送信
+const sendNewHome = async () => {
+  // 褒め言葉に関する条件を満たしたらPOST
+  const newSentence = elems.input.value.trim();
+  if (!newSentence) return;
+
+  if (!newSentence) {
+    alert("褒め言葉を入力してください。");
+    return;
+  }
```

Comment:
二重のreturnになってしまい、期待されるErrorが出ないよ！
```suggestion
  if (!newSentence) {
    alert("褒め言葉を入力してください。");
    return;
  }
```